### PR TITLE
Add eslint rules for comments

### PR DESCRIPTION
**Pull request by takameyer**
## What, How & Why?
* Space after `//`
* Comment blocks start capitalized

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated

### CODE REVIEW

File: packages/realm/src/Realm.ts [hunk 1]
```diff
@@ -194,7 +194,7 @@ type RealmSchemaExtra = Record<string, ObjectSchemaExtra | undefined>;
 type ObjectSchemaExtra = {
   constructor?: RealmObjectConstructor;
   defaults: Record<string, unknown>;
-  // objectTypes: Record<string, unknown>;
+  // ObjectTypes: Record<string, unknown>;
```
Comment (kraenhansen) [inline packages/realm/src/Realm.ts:197]: Admittedly, this line should probably be deleted, but I can see how the `capitalized-comments` rule would be annoying when commenting out code while developing and having format-on-save mess it up like this.

File: packages/realm/src/Configuration.ts [hunk 2]
```diff
@@ -59,7 +59,7 @@ export type ConfigurationWithoutSync = BaseConfiguration & {
 
 export type Configuration = ConfigurationWithSync | ConfigurationWithoutSync;
 
-// export type PartitionValue = string | number | null | ObjectId | UUID;
+// Export type PartitionValue = string | number | null | ObjectId | UUID;
```
Comment (kraenhansen) [inline packages/realm/src/Configuration.ts:62]: Similar to my other comment

File: integration-tests/tests/src/utils/benchmark.ts [hunk 3]
```diff
@@ -60,7 +60,7 @@ export function describePerformance(title: string, parameters: PerformanceTestPa
     before(parameters.before);
     itPerforms(parameters.benchmarkTitle, parameters.test);
     after(function (this: BenchmarkContext) {
-      // console.log(this.summary);
+      // Console.log(this.summary);
```
Comment (kraenhansen) [inline integration-tests/tests/src/utils/benchmark.ts:63]: Another bad fix.

File: integration-tests/tests/src/typings.d.ts [hunk 4]
```diff
@@ -174,7 +174,7 @@ type RealmContext = {
 type RealmObjectContext<T = Record<string, unknown>> = {
   object: RealmObject & T;
 } & RealmContext;
-// type BenchmarkContext = {
+// Type BenchmarkContext = {
 //   result: BenchmarkResult;
 // } & Mocha.Context;
```
Comment (kraenhansen) [inline integration-tests/tests/src/typings.d.ts:179]: Here as well ..

File: integration-tests/tests/src/typings.d.ts [hunk 5]
```diff
@@ -20,7 +20,7 @@ type RealmObject = import("realm").Object;
 type App = import("realm").App;
 type User = import("realm").User;
 type Configuration = import("realm").Configuration;
-//type BenchmarkResult = import("@thi.ng/bench").BenchmarkResult;
+// Type BenchmarkResult = import("@thi.ng/bench").BenchmarkResult;
```
Comment (kraenhansen) [inline integration-tests/tests/src/typings.d.ts:23]: And here.

**Review by kraenhansen — COMMENTED** [review]
There's a couple of places where commented out source-code gets updated. Admittedly that code should probably be deleted, but I can see how the `capitalized-comments` rule combined with format-on-save would be super annoying at times.

File: .eslintrc.json [hunk 6]
```diff
@@ -6,7 +6,9 @@
     "./header.eslintrc"
   ],
   "rules": {
-    "no-restricted-globals": ["error", "globalThis"]
+    "no-restricted-globals": ["error", "globalThis"],
+    "spaced-comment": ["error", "always", {"exceptions":["/"]}],
+    "capitalized-comments": ["error", "always", {"ignoreConsecutiveComments": true }]
```
Comment (kraenhansen) [inline .eslintrc.json:11]: I'd say "warn" would be good enough here.

Comment (elle-j) [inline .eslintrc.json:11]: I saw e.g. `// tslint:disable-next-line` in some tests and not sure if the capitalization will invalidate that. If so, we could add it to the [ignorePattern](https://eslint.org/docs/latest/rules/capitalized-comments#ignorepattern). But maybe those tslints can even be removed?
```suggestion
    "capitalized-comments": ["error", "always", {
      "ignorePattern": "tslint",
      "ignoreConsecutiveComments": true
    }]
```

**Review by elle-j — APPROVED** [review]

File: .eslintrc.json [hunk 7]
```diff
@@ -6,7 +6,9 @@
     "./header.eslintrc"
   ],
   "rules": {
-    "no-restricted-globals": ["error", "globalThis"]
+    "no-restricted-globals": ["error", "globalThis"],
+    "spaced-comment": ["error", "always", {"exceptions":["/"]}],
```
[the remark this answers is not in the document]
Comment (takameyer) [reply .eslintrc.json:10]: Good idea on `balanced`.  The expection for `/` is still needed due to our headers.  Granted, we could reformat or header to be similar to our other comment blocks

**Comment by takameyer** [comment]
Thanks for the comments so far.  I have attempted some of the suggestions, but haven't gotten it in a state where I'm really happy with the results.  Our lovely header is making things a bit difficult.  Perhaps we should reformat it, although I have also gone down this road and haven't found a good alternative yet. 
Honestly, I have spent more time on this than intended 😄.  I'm going to leave this open for now and get back to it when as time permits.

File: .eslintrc.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (elle-j) [reply .eslintrc.json:10]: Oh yeah, then we could use `exceptions` rather than `markers` if you'd like 🙂

**Comment by takameyer** [comment]
I will most likely never get back to this.  More effort than it was worth.

**Pull request closed without merging**